test/health_failures_test_case.py

```diff
@@ -0,0 +1,107 @@
+                    'AttributeName': 'SentTimeMessageId',
+                    'AttributeType': 'S'
+                }
+            ],
+            'ProvisionedThroughput': {
+                'ReadCapacityUnits': 5,
+                'WriteCapacityUnits': 5
+            }
+        }
+        database.create_table(**table_settings)
+        with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
+            try:
+                yield
+            finally:
+                table = database.Table(config.dynamo_failure_message_table_name)
+                table.delete()
+
+    @mock_sts
+    @mock_sqs
+    @mock_dynamodb2
+    def test_failures_endpoint(self):
+        from more_itertools import chunked
+        indexer_app = load_app_module('indexer')
+        sqs = boto3.resource('sqs', region_name='us-east-1')
+        sqs.create_queue(QueueName=config.fail_queue_name)
+        fail_queue = sqs.get_queue_by_name(QueueName=config.fail_queue_name)
+        with ResponsesHelper() as helper:
+            helper.add_passthru(self.base_url)
+            # The 4th sub-test checks if the indexer lambda can write more than 1 batch of messages to dynamodb.
+            # The max number of messages in a batch is 10 and this sub-test populates the queue with 11 messages.
+            for bundle_notification_count, other_count in ((0, 0), (1, 0), (0, 1), (10, 1)):
+                with self._make_database():
+                    bundle_notifications = [{
+                        'action': 'add',
+                        'notification': IndexerTestCase._make_fake_notification((str(uuid4()),
+                                                                                 '2019-10-14T113344.698028Z'))
+                    } for i in range(bundle_notification_count)]
+                    other_notifications = [{
+                        'other': 'notification'
+                    } for i in range(other_count)]
+                    for batch in chunked(bundle_notifications + other_notifications, 10):
+                        items = [{'Id': str(i), 'MessageBody': json.dumps(message)} for i, message in enumerate(batch)]
+                        fail_queue.send_messages(Entries=items)
+                    expected_response = {
+                        "failed_bundle_notifications": bundle_notifications,
+                        "other_failed_messages": other_count
+                    }
+                    with self.subTest(bundle_notification_count=bundle_notification_count,
+                                      document_notification_count=other_count):
+                        indexer_app.retrieve_failure_messages(MagicMock(), MagicMock())
+                        response = requests.get(self.base_url + '/health/failures')
+                        self.assertEqual(200, response.status_code)
+                        actual_response = response.json()
+                        self.assertEqual(expected_response.keys(), actual_response.keys())
```

Comment:
Why isn't this just `assertEqual(expected_response, actual_response)`?

Reply:
`assertEqual(expected_response, actual_response)` would fail because the items in `failed_bundle_notifications` list in `actual_response` are not in the same order as the `failed_bundle_notifications` list in `expected_response`. 

The `failed_bundle_notifications` list populated from the SQS fail queue and these messages can't always be retrieved from the SQS in the same order the messages were put into the queue. Instead, the content of the 'failed_bundle_notifications' list is tested in self.assertCountEqual(expected_response['failed_bundle_notifications'], actual_response['failed_bundle_notifications']).

Reply:
I was afraid you'd give this explanation. Please make this an order-insensitive comparison.